src/components/Swap/buttonState.ts

```diff
@@ -1,96 +1,111 @@
 import { isEmpty } from 'ramda';
-import { ErgoBox } from 'ergo-dex-sdk/build/module/ergo';
+import { AssetAmount, ErgoBox } from 'ergo-dex-sdk/build/module/ergo';
+
+const SUBMIT_TEXT = 'Submit';
 
 export enum States {
-  NEED_TO_CONNECT_WALLET = 'NEED_TO_CONNECT_WALLET',
+  LOADING = 'LOADING',
   SELECT_A_TOKEN = 'SELECT_A_TOKEN',
   NEED_TO_ENTER_AMOUNT = 'NEED_TO_ENTER_AMOUNT',
   NEED_TO_CHOOSE_ADDRESS = 'NEED_TO_CHOOSE_ADDRESS',
   PENDING_TRANSACTION = 'PENDING_TRANSACTION',
   UTXOS_IS_EMPTY = 'UTXOS_IS_EMPTY',
+  INSUFFICIENT_INPUT_TOKEN_BALANCE = 'INSUFFICIENT_INPUT_TOKEN_BALANCE',
   SUBMIT = 'SUBMIT',
 }
 
 interface ButtonStateDependencies {
-  isWalletConnected: boolean;
-  inputAssetId?: string;
-  outputAssetId?: string;
+  inputAssetAmount?: AssetAmount;
+  outputAssetAmount?: AssetAmount;
   inputAmount: string;
   outputAmount: string;
   chosenAddress: string | undefined;
   utxos: ErgoBox[] | undefined;
-  availableInputAmount: bigint;
+  availableAmount: { input: bigint; output: bigint };
 }
 
 interface ButtonState {
   isDisabled: boolean;
   text: string;
 }
 
+const makeState = (text: string): ButtonState => {
+  if (text === SUBMIT_TEXT) {
+    return { text, isDisabled: false };
+  }
+
+  return { text, isDisabled: true };
+};
+
```

Comment:
This function reduces the declarativity of the explicit finite state machine pattern. I think there is no problem to write verbose state config. It is plain and simple. This function feels like a zip-archiver it reduces total size but makes it harder to read. For example this condition `text === SUBMIT_TEXT` not very intuitive

Reply:
agree.